doc/src/manual/interfaces.md

```diff
@@ -739,3 +739,67 @@ yields another `SparseVecStyle`, that its combination with a 2-dimensional array
 yields a `SparseMatStyle`, and anything of higher dimensionality falls back to the dense arbitrary-dimensional framework.
 These rules allow broadcasting to keep the sparse representation for operations that result
 in one or two dimensional outputs, but produce an `Array` for any other dimensionality.
+
+## [IO Stream](@id man-interface-iostream)
+
+| Required methods                          | Brief description                                                                               |
+|:----------------------------------------- |:----------------------------------------------------------------------------------------------- |
+| `unsafe_read(io, ::Ptr{UInt8}, ::UInt)`   | Copy bytes from the `IO` to a pointer. |
+| `unsafe_write(io, ::Ptr{UInt8}, ::UInt)`  | Copy bytes from a pointer to the `IO`. |
+| `eof(io)`                                 | Whether the IO stream is at the end.   |
+
+| Optional methods                          | Brief description                                                                               |
+|:----------------------------------------- |:----------------------------------------------------------------------------------------------- |
+| `read(io, ::Type{UInt8})`                 | Read a byte from the stream.               |
+| `write(io, ::UInt8)`                      | Write a byte to the stream.                |
+| `close(io)`                               | Close the stream.                          |
+| `seek(io, ::Integer)`                     | Seek to a specific position.               |
+| `position(io)`                            | Return the current position of the stream. |
+| `seekstart(io)`                           | Seek to beginning of stream.               |
+| `seekend(io)`                             | Seek to the end of the stream.             |
+
+
+To implement an `IO` object it is required to define the low-level methods `unsafe_read`
+and `unsafe_write` which enable copying of data between the `IO` and a location given by a
+pointer.  Additionally, `eof` is required and should return `true` if reading from the
+`IO` is valid, else `false`.
```

Comment:
```suggestion
To implement an `IO` object it is required to define the low-level methods for
byte `read` and `write` which enable reading and writing from the `IO`.
Additionally, `eof` is required and should return `true` if a `read` from the
`IO` will return at least one byte, else `false` if it will not return any more bytes.
```